usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -89,31 +102,107 @@ def database_data_layer(self):
 
         return queryset, values
 
+    def apply_elasticsearch_aggregations(self, search: Search) -> None:
+        """
+        Takes in an instance of the elasticsearch-dsl.Search object and applies the necessary
+        aggregations in a specific order to get expected results.
+        """
+        interval = "year" if self.group == "fiscal_year" else self.group
+
+        # The individual aggregations that are needed; with two different sum aggregations to handle issues with
+        # summing together floats.
+        group_by_time_period_agg = A(
+            "date_histogram", field="fiscal_action_date", interval=interval, format="yyyy-MM-dd"
+        )
+        sum_as_cents_agg = A(
```

Comment:
I'm not completely following the justification for counting by pennies. Doesn't the `scaled_float` data type prevent floating point rounding errors?

Reply:
There were still issues with floating point numbers because while scaled float with a factor of 100 stores its values as `203` when it is really `2.03` that value for aggregations and queries is represented as the `2.03`. So the scaled floats [are being summed together in the aggregation as regular floats](https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html#scaled-float-params). Noticed the issue during the prototype phase.

Reply:
Mad world... thanks

Reply:
I wanted to add that I also found this: https://github.com/elastic/elasticsearch/pull/37169. Once we upgrade to a new version of Elasticsearch the issue might actually go away. Apparently there were some precision errors found with `scaled_float` and it was fixed as part of `6.7`.